float64 theta0
float64 theta1
float64 theta2
float64 theta3
float64 theta4
float64 theta5
---
float64 x
float64 y
float64 z